lib/async_hooks.js

```diff
@@ -204,6 +205,105 @@ class AsyncResource {
   }
 }
 
+const locals = [];
```

Comment:
A Set seems more appropriate than an array here. Is it like this for performance reasons?

Reply:
Yes, @mcollina advised for a `for(;;)` loop out of performance concern